ziti/edge/types.go

```diff
@@ -170,3 +181,188 @@ func ApiResponseDecode(data interface{}, resp io.Reader) (*ApiResponseMetadata,
 
 	return apiR.Meta, nil
 }
+
+type InteceptDialOptions struct {
+	ConnectTimeoutSeconds *int
+	Identity              *string
+}
+
+type PortRange struct {
+	Low  uint16
+	High uint16
+}
+
+type InterceptV1Config struct {
+	Addresses   []ZitiAddress
+	PortRanges  []*PortRange
+	Protocols   []string
+	SourceIp    *string
+	DialOptions *InteceptDialOptions `json:"dialOptions"`
+	Service     *Service
+}
+
+type DomainName string
+
+func (dn DomainName) Match(hostname string) int {
+	if len(dn) < 3 {
+		return -1
+	}
+
+	if dn[0] == '*' {
+		domain := string([]byte(dn)[1:])
+		if strings.HasSuffix(hostname, domain) {
+			return len(hostname) - len(domain)
+		} else {
+			return -1
+		}
+	} else {
+		if hostname == string(dn) {
```

Comment:
why the if, why not return the comparison of hostname to string(dn)? if there's a reason, I'd love to see a comment explaining why

Reply:
not sure what you mean by comparison. return is an value is an int -- -1 not matching, or the score.